script_test.go

```diff
@@ -1850,6 +1850,82 @@ func TestReadReturnsErrorGivenReadErrorOnPipe(t *testing.T) {
 	}
 }
 
+func TestEncodeBase64(t *testing.T) {
+	t.Parallel()
+	tcs := []struct {
+		name string
+		s    string
+		want string
+	}{
+		{
+			name: "empty string",
+			s:    "",
+			want: "\n",
+		},
+		{
+			name: "single line string",
+			s:    "hello world",
+			want: "aGVsbG8gd29ybGQ=\n",
+		},
+		{
+			name: "multi line string",
+			s:    "hello\nthere\nworld\n",
+			want: "aGVsbG8KdGhlcmUKd29ybGQK\n",
+		},
+	}
+
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			s, err := script.Echo(tc.s).EncodeBase64().String()
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			if s != tc.want {
+				t.Errorf("want %s, got %s", tc.want, s)
+			}
+		})
+	}
+}
+
+func TestDecodeBase64(t *testing.T) {
+	t.Parallel()
+	tcs := []struct {
+		name string
+		s    string
+		want string
+	}{
+		{
+			name: "empty string",
+			s:    "",
+			want: "\n",
+		},
+		{
+			name: "single line output",
+			s:    "aGVsbG8gd29ybGQ=",
+			want: "hello world\n",
+		},
+		{
+			name: "multi line output",
```

Comment:
Shall we use the same set of cases for testing both `Encode` and `Decode`? Also, what about a round-trip test?

Reply:
@bitfield would you be able to clarify? do you want one test for both methods? and yes, i can add a round trip test!

Reply:
We don't have to combine the two tests into one; instead, we could move the test cases outside the function, so that both tests can share it.

```go
var base64Cases = ...

func TestDecodeBase64...
```